### PR TITLE
Added conditional rules around the "base may be wrong" warnings for stats.

### PR DESCRIPTION
**Pull request by ryan-rushton**
I have another branch up where I distinguish between conditional and negative stats but I think it might be a bit much/not really be useful information.

### CODE REVIEW

**Comment by delphiactual** [comment]
In the other branch `_context` should be used not interpolation e.g.

```
"TotalIncorrectWarning_negative": "Due to negative stat modifiers, DIM cannot determine the base value of {{stats}}. This is a limitation of the information sent by Bungie.net."
"TotalIncorrectWarning_conditional": "Due to conditional stat modifiers, DIM cannot determine the base value of {{stats}}. This is a limitation of the information sent by Bungie.net."
```

```
// t('Stats.TotalIncorrectWarning_conditional') t('Stats.TotalIncorrectWarning_negative')

t('Stats.TotalIncorrectWarning', {
        context: incorrectStats?.conditional.length ? 'conditional' : incorrectStats?.negative.length ? 'negative' : '',
        stats: incorrectStats.conditional.join('/'),
      })
```

File: config/i18n.json [hunk 1]
```diff
@@ -874,7 +874,7 @@
     "TierProgress": "T{{tier}} {{statName}} ({{progress}}/60 for T{{nextTier}})\n",
     "TierProgress_Max": "T{{tier}} {{statName}} ({{progress}}/300)\n",
     "Total": "Total",
-    "TotalIncorrectWarning": "Due to negative stat modifiers, DIM cannot determine the base value of {{stats}}. This is a limitation of the information sent by Bungie.net."
+    "TotalIncorrectWarning": "Due to mods with negative stat modifiers or conditional stat activations, DIM cannot determine the base value of {{stats}}. This is a limitation of the information sent by Bungie.net."
```
Comment (sundevour) [inline config/i18n.json:877]:
```suggestion
    "TotalIncorrectWarning": "Due to mods with negative stat modifiers or conditional stat effects, DIM cannot determine the base value of {{stats}}. This is a limitation of the information sent by Bungie.net."
```

File: config/i18n.json [hunk 2]
```diff
@@ -572,7 +572,7 @@
     "SelectPerks": "Select Perks",
     "SelectPower": "Minimum Power",
     "SelectPowerDescription": "Sets a minimum power level for the entire loadout (not individual pieces)",
-    "StatIncorrectWarning": "Please remove any mods with negative stat values to get accurate results.\n\nDue to negative stat modifiers, DIM cannot determine the base value of {{stats}} on some included items. This is a limitation of the information sent by Bungie.net.",
+    "StatIncorrectWarning": "Please remove any mods with negative stat values or that are conditionally applied to get accurate results.\n\nDue to mods with negative stat modifiers or conditional stat activations, DIM cannot determine the base value of {{stats}} on some included items. This is a limitation of the information sent by Bungie.net.",
```
Comment (sundevour) [inline config/i18n.json:575]:
```suggestion
    "StatIncorrectWarning": "To get accurate results, please remove any mods with negative stat values or conditional effects.\n\nDue to mods with negative stat modifiers or conditional stat effects, DIM cannot determine the base value of {{stats}} on some included items. This is a limitation of the information sent by Bungie.net.",
```

**Comment by ryan-rushton** [comment]
> In the other branch `_context` should be used not interpolation e.g.
> 
> ```
> "TotalIncorrectWarning_negative": "Due to negative stat modifiers, DIM cannot determine the base value of {{stats}}. This is a limitation of the information sent by Bungie.net."
> "TotalIncorrectWarning_conditional": "Due to conditional stat modifiers, DIM cannot determine the base value of {{stats}}. This is a limitation of the information sent by Bungie.net."
> ```
> 
> ```
> // t('Stats.TotalIncorrectWarning_conditional') t('Stats.TotalIncorrectWarning_negative')
> 
> t('Stats.TotalIncorrectWarning', {
>         context: incorrectStats?.conditional.length ? 'conditional' : incorrectStats?.negative.length ? 'negative' : '',
>         stats: incorrectStats.conditional.join('/'),
>       })
> ```

Cool, I thought there was a better way to do that. If we end up going with that option I will fix it up.

File: config/i18n.json (hunk 1 above)
Comment (ryan-rushton) [reply config/i18n.json:877]: Yeah it probably is, will change.

File: config/i18n.json (hunk 2 above)
Comment (ryan-rushton) [reply config/i18n.json]: Done

**Comment by ryan-rushton** [comment]
Merging this for now but I won't kill my other branch just yet, in the event we think it needs more distinction between the two warnings.

**Pull request merged**